src/main/java/com/stuypulse/robot/util/LEDController.java

```diff
@@ -0,0 +1,134 @@
+import com.stuypulse.stuylib.network.limelight.Limelight;
+
+import edu.wpi.first.wpilibj.PWMSparkMax;
+import edu.wpi.first.wpilibj.Timer;
+
+public class LEDController {
+
+    private static PWMSparkMax controller;
+    private static Shooter shooter;
+    private static Chimney chimney;
+    private static Drivetrain drivetrain;
+
+    public LEDController(int port, Shooter shooter, Chimney chimney, Drivetrain drivetrain) {
+        controller = new PWMSparkMax(port);
+        this.shooter = shooter;
+        this.chimney = chimney;
+        this.drivetrain = drivetrain;
+    }
+
+    public static void setValue(double value) {
+        controller.set(value);
+    }
+
+    public static double getValue() {
+        return controller.get();
+    }
+
+    enum Color {
+        WHITE_SOLID,  // Shoot from initation line
+        PINK_SOLID,   // Shoot from trench
+        RED_SOLID,    // Shoot from far
+        WHITE_PULSE,  // Charging shot from initation line
+        PINK_PULSE,   // Charging shot from trench
+        RED_PULSE,    // Charging shot from far
+        YELLOW_SOLID, // Aligning
+        LIME_FLASH,   // Aligned
+        BLUE_FLASH,   // Ball detected in chute
+        OFF
+    }
+
+    // Switches the color on the thread you call it from
+    private static void setColorOnThread(Color color) {
+        switch (color) {
+            case WHITE_SOLID:
+                setValue(0.93);
+                break;
+            case PINK_SOLID:
+                setValue(0.57);
+                break;
+            case RED_SOLID:
+                setValue(0.61);
+                break;
+            case WHITE_PULSE:
+                setValue(0.93);
+                Timer.delay(0.5);
+                setValue(0.99);
+                Timer.delay(0.1);
+                break;
+            case PINK_PULSE:
+                setValue(0.57);
+                Timer.delay(0.5);
+                setValue(0.99);
+                Timer.delay(0.1);
+                break;
+            case RED_PULSE:
+                setValue(0.61);
+                Timer.delay(0.5);
+                setValue(0.99);
+                Timer.delay(0.1);
+            case YELLOW_SOLID:
+                setValue(0.69);
+                break;
+            case LIME_FLASH:
+                if (getValue() != 0.99) {
+                    setValue(0.73);
+                    Timer.delay(0.5);
+                    setValue(0.99); 
+                } 
+                break;
+            case BLUE_FLASH:
+                if (getValue() != 0.99) {
+                    setValue(0.87);
+                    Timer.delay(0.5);
+                    setValue(0.99);
+                } 
+                break;
+            case OFF:
+                setValue(0.99);
+                break;
+        }
+    }
+
+    // Set color on new thread
+    public static void setColor(Color color) {
+        new Thread(() -> setColorOnThread(color)).start();
```

Comment:
Wouldn't this start a new thread whenever you want to set color? I think it would be better to start one thread in robot.java that runs control leds instead.